pkg/pss/pss_test.go

```diff
@@ -0,0 +1,223 @@
+// Copyright 2020 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package pss_test
+
+import (
+	"context"
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/ethersphere/bee/pkg/pss"
+	"github.com/ethersphere/bee/pkg/storage"
+	"github.com/ethersphere/bee/pkg/storage/mock"
+	"github.com/ethersphere/bee/pkg/swarm"
+	"github.com/ethersphere/bee/pkg/tags"
+	"github.com/ethersphere/bee/pkg/trojan"
+)
+
+// TestTrojanChunkRetrieval creates a trojan chunk
+// mocks the localstore
+// calls pss.Send method and verifies it's properly stored
+func TestTrojanChunkRetrieval(t *testing.T) {
+	var err error
+	ctx := context.TODO()
+	testTags := tags.NewTags()
+
+	localStore := mock.NewTagsStorer(testTags)
+	pss := pss.NewPss(localStore, testTags)
+
+	target := trojan.Target([]byte{1}) // arbitrary test target
+	targets := trojan.Targets([]trojan.Target{target})
+	payload := []byte("RECOVERY CHUNK")
+	topic := trojan.NewTopic("RECOVERY TOPIC")
+
+	// call Send to store trojan chunk in localstore
+	if _, err = pss.Send(ctx, targets, topic, payload); err != nil {
+		t.Fatal(err)
+	}
+
+	var chunkAddress swarm.Address
+	// this code iterates over the localstore
+	// this will get the chunk that was stored in pss.Send
+	for po := uint8(0); po <= swarm.MaxPO; po++ {
+		last, err := localStore.LastPullSubscriptionBinID(po)
+		if err != nil {
+			t.Fatal(err)
+		}
+		if last == 0 {
+			continue
+		}
+		// iter for chunk in localstore
+		ch, _, _ := localStore.SubscribePull(context.Background(), po, 0, last)
+		for c := range ch {
+			chunkAddress = c.Address
+			break
+		}
+	}
+
+	// verify store, that trojan chunk has been stored correctly
+	var storedChunk swarm.Chunk
+	if storedChunk, err = localStore.Get(ctx, storage.ModeGetRequest, chunkAddress); err != nil {
+		t.Fatal(err)
+	}
+
+	// create a stored chunk artificially
```

Comment:
... if you had that address check interface probably you wouldn't need this code.